tests/functional/io_apis_test.c

```diff
@@ -0,0 +1,831 @@
+// Copyright 2018-2022 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+/*
+ * ----------------------------------------------------------------------------
+ * splinter_io_apis_test.c --
+ *
+ * This test exercises core IO interfaces that are used by SplinterDB
+ * to verify that these interfaces work as expected when executed:
+ *
+ * - Using threads, with Splinter using default malloc()-memory allocation
+ *
+ * The test design is quite simple:
+ *
+ * - Starting with a default IO configuration, create a device of some size
+ *   and using Sync IO interfaces driven by the main program, fill the
+ *   device out with some known data. Read back each page worth of data from
+ *   the file using sync-read and verify correctness. This case covers the
+ *   correctness check for basic sync read/write APIs driven by the main thread.
+ *
+ * - Using the data generated above, fire-up n-threads. Assign contiguous
+ *   sections of the file to each thread. Each thread verifies previously
+ *   written data using sync-read. Then, each thread writes new data to its
+ *   section using sync-writes, and verifies using sync-reads.
+ * ----------------------------------------------------------------------------
+ */
+
+#include "platform.h"
+#include "config.h"
+#include "io.h"
+#include "trunk.h" // Needed for trunk_get_scratch_size()
+#include "task.h"
+
+/*
+ * Structure to package arguments needed by test-case functions. This packaging
+ * allows to pass this set of args around. These are then supplied by worker
+ * functions invoked by pthreads to invoke the work-horse function(s).
+ */
+typedef struct io_test_fn_args {
+   platform_heap_id    arg_hid;
+   io_config          *arg_io_cfgp;
+   platform_io_handle *arg_io_hdlp;
+   io_context_t        arg_io_ctxt_t; // Expected opaque context handle
+   task_system        *arg_tasks;
+   uint64              arg_start_addr;
+   uint64              arg_end_addr;
+   char                arg_stamp_char;
+   platform_thread     arg_thread;
+} io_test_fn_args;
+
+/* Whether to display verbose-progress from each thread's activity */
+bool Verbose_progress = FALSE;
+
+/*
+ * Different test cases in this test drive multiple threads each doing one
+ * type of activity. Declare the interface of such thread handler functions.
+ */
+typedef void (*test_io_thread_hdlr)(void *arg);
+
+/* SplinterDB device size; small one is good enough for IO APIs testing */
+#define SPLINTER_DEVICE_SIZE_MB 128
```

Comment:
Maybe just `DEVICE_SIZE_MB` since there's nothing about splinter in here?

Reply:
Done.